WorkloadManagementSystem/Utilities/PilotCStoJSONSynchronizer.py

```diff
@@ -365,9 +367,19 @@ def _upload(self, pilotDict=None, filename='', pilotScript=''):
       with open(pilotScript, "rb") as psf:
         script = psf.read()
       data = {'filename': filename, 'data': script}
-
+    
+    bd = BundleDeliveryClient()
```

Comment:
This will (re)create the bundle, right?
Don't you simply need `Locations.getCAsLocation`?

Reply:
No, this is the recommended way to use it.
BundleDeliveryClient is using the CAs files in the machine, create a single file. But in case the CAs is not found the machine will download it from the machine otherwise use the certifi

Reply:
OK. Please check what's failing in this and in the other PR in Travis.

Reply:
I am not 100% sure I understand. When should we use `getCAsLocation`, when should we use `BundleDeliveryClient` ? Is this documented somewhere ?

Reply:
It is not documented. BundleDeloveryClient is using getCAsLocation to generate the CAs file.
Most of the case you can not give a directory, which contains the certificates to a certain client , you have to provide a file, which contains all CAs.  Because most of the libraries does not work with a directory which contains a list of files.

Reply:
That is not true, most of the library can take either the path or the file. 
I believe that the cleanest would be to have getCALocation take an option to return either the bundled file or the directory. But it should not be the responsibility of each and every client to download and bundle things

Reply:
It is not downloading by default, just create a single file. Elasticsearch does not accept directory, tornado as well. I do not see any problem to create a single file using getCAsLocation and this 100% will work.

Reply:
It is safer, but if you prefer I can use the method as you prefer. 
Note: I am not going to check if the very old version of request can accept a directory (the version which is used by DIRACOS accept a directory as well, but I am not sure the version what we are using in prod still do)

Reply:
I do not think it is safer. Just imagine that you have two clients on teh same machine using the same call. They will both try to create this single file, and step on each other foot. 
The cleanest method is to have a cron job creating this single file on all VOBoxes, an then have `getCALocations` return either the directory or the file

Reply:
This "cron" job is called CAUpdateAgent. It can be updated itself to update CA's (possibly from multiple sources) and create single CA file

Reply:
It makes sense that there is an agent, although I strongly believe that since we have a cron job to fetch the CAs and CRLs in the first place, we should stick with cron job to generate that one file

Reply:
We need to make up our mind on how to handle this business once for all. It makes sense to have each every server with the same set of CAs/CRLs/bundle whatever, so if this is the case the same cron job or agent has to be part of the server installation "instructions". Let's discuss this at the coming BiLD.